lib/ret/media_resolver.ex

```diff
@@ -426,46 +461,61 @@ defmodule Ret.MediaResolver do
     end
   end
 
-  def resolved(:error) do
-    nil
-  end
+  def resolved(:error), do: nil
+  def resolved(%URI{} = uri), do: %Ret.ResolvedMedia{uri: uri}
+  def resolved(%URI{} = uri, meta), do: %Ret.ResolvedMedia{uri: uri, meta: meta}
 
-  def resolved(%URI{} = uri) do
-    %Ret.ResolvedMedia{uri: uri}
-  end
-
-  def resolved(%URI{} = uri, meta) do
-    %Ret.ResolvedMedia{uri: uri, meta: meta}
-  end
+  def resolved(%URI{} = uri, %URI{} = audio_uri, meta),
+    do: %Ret.ResolvedMedia{uri: uri, audio_uri: audio_uri, meta: meta}
 
   defp module_config(key) do
     Application.get_env(:ret, __MODULE__)[key]
   end
 
-  defp ytdl_resolution(%MediaResolverQuery{low_resolution: true}), do: "480"
-  defp ytdl_resolution(_query), do: "720"
+  defp ytdl_resolution(%MediaResolverQuery{quality: :low}), do: "[height<=480]"
+  defp ytdl_resolution(%MediaResolverQuery{quality: :low_360}), do: "[height<=1440]"
+  defp ytdl_resolution(%MediaResolverQuery{quality: :high_360}), do: "[height<=2160]"
```

Comment:
Hmm, would this just return no results if you requested a `high_360` but only 1440 was available? Bit confusing how this interacts with `best` and `bestvideo`

Reply:
It's `<=`, so it will prefer the highest resolution video that is at most 2160 I assume.